duckdb-vortex/src/vortex_duckdb_extension.cpp

```diff
@@ -1,52 +1,70 @@
 #define DUCKDB_EXTENSION_MAIN
 
-#include "vortex_duckdb_extension.hpp"
 #include "duckdb/common/exception.hpp"
+#include "duckdb/common/helper.hpp"
+#include "duckdb/common/multi_file_reader.hpp"
+#include "duckdb/common/multi_file_reader_function.hpp"
 #include "duckdb/function/scalar_function.hpp"
+#include "duckdb/function/table_function.hpp"
 #include "duckdb/main/extension_util.hpp"
+#include "duckdb/parser/parsed_data/create_table_function_info.hpp"
+
+#include "vortex_duckdb_extension.hpp"
 
 extern "C" {
 const char *vortex_duckdb_hello();
 }
 
-namespace duckdb {
+#ifndef DUCKDB_EXTENSION_MAIN
+#error DUCKDB_EXTENSION_MAIN not defined
+#endif
 
-inline void VortexDuckdbScalarFun(DataChunk &args, ExpressionState &state, Vector &result) {
-	auto &name_vector = args.data[0];
-	UnaryExecutor::Execute<string_t, string_t>(name_vector, result, args.size(), [&](string_t name) {
-		auto c_str = vortex_duckdb_hello();
-		auto str = std::string(c_str) + name.GetString() + " 🐥";
-		return StringVector::AddString(result, str);
-	});
-}
+class VortexFileReaderOptions : public duckdb::BaseFileReaderOptions {
+public:
+	bool file_reader_flag {false};
+};
 
-static void LoadInternal(DatabaseInstance &instance) {
-	// Register a dummy table function
-	auto vortex_duckdb_scalar_function =
-	    ScalarFunction("vortex_hello", {LogicalType::VARCHAR}, LogicalType::VARCHAR, VortexDuckdbScalarFun);
-	ExtensionUtil::RegisterFunction(instance, vortex_duckdb_scalar_function);
-}
+struct VortexReadBindData : public duckdb::TableFunctionData {
+	bool read_bind_data_flag {false};
+};
 
-void VortexDuckdbExtension::Load(DuckDB &db) {
-	LoadInternal(*db.instance);
-}
+struct VortexInitData : public duckdb::GlobalTableFunctionState {
+	std::atomic<bool> done {false};
 
-std::string VortexDuckdbExtension::Name() {
-	return "vortex_duckdb";
-}
+	static duckdb::unique_ptr<GlobalTableFunctionState> Create() {
+		return duckdb::make_uniq<VortexInitData>();
+	}
+};
 
-std::string VortexDuckdbExtension::Version() const {
-#ifdef EXT_VERSION_VORTEX_DUCKDB
-	return EXT_VERSION_VORTEX_DUCKDB;
-#else
-	return "";
-#endif
-}
+// Vortex file info implementation to integrate with DuckDBs MultiFileReader.
+struct VortexMultiFileInfo {
+	static void GetBindInfo(const duckdb::TableFunctionData &bind_data_p, duckdb::BindInfo &info) {
+		throw std::runtime_error("called vortex get bind info func");
+		// auto &bind_data = bind_data_p.Cast<VortexReadBindData>();
+		// info.type = duckdb::ScanType::EXTERNAL;
+		// info.InsertOption("setting", duckdb::Value::BOOLEAN(bind_data.read_bind_data_flag));
```

Comment:
what is this for?

Reply:
Actually I think I'll drop this in the next PR. Analog to https://github.com/duckdb/duckdb-avro/blob/main/src/avro_extension.cpp it seems possible to get going without implementing an API like https://github.com/duckdb/duckdb/blob/main/extension/parquet/parquet_extension.cpp#L88.